internal/worker.go

```diff
@@ -333,8 +334,11 @@ func NewWorker(
 	domain string,
 	taskList string,
 	options WorkerOptions,
-) *aggregatedWorker {
-	return newAggregatedWorker(service, domain, taskList, options)
+) (*aggregatedWorker, error) {
+	if err := options.Validate(); err != nil {
+		return nil, fmt.Errorf("worker options validation error: %w", err)
+	}
+	return newAggregatedWorker(service, domain, taskList, options), nil
```

Comment:
might be better to put this inside newAggregatedWorker tbh, since it modifies the options after this.

validating before that makes plenty of sense, but it does feel a bit odd to *not* validate after it, in case we messed up the `AugmentWorkerOptions` somehow.